### PR TITLE
must-gather: initial commit

### PR DESCRIPTION
**Pull request by nbalacha**
The lvm must-gather will capture lvm related information
from an OCP cluster.

Signed-off-by: N Balachandran <nibalach@redhat.com>

### CODE REVIEW

File: must-gather/Centos-Base.repo [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[centos]
+name=CentOS-7
```
Comment (jmolmo) [inline must-gather/Centos-Base.repo:2]: This tool is completely linked to RH. Why not to use an UBI image?

File: must-gather/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+OCS must-gather
+=================
+
+`lvm-must-gather` is a tool built on top of [OpenShift must-gather](https://github.com/openshift/must-gather)
```
Comment (jmolmo) [inline must-gather/README.md:4]: The name of this tool is OCS must-gather or "lvm must gather"?

**Review by jmolmo — COMMENTED** [review]
If we are going to use this tool from inside topolvm, I think we need a better way to retrieve the information directly from the "code", I mean.. we will need to do several changes  to use the tool as library.
In other side, it is for openshift... now we have a very good opportunity to make it more generic, allowing to be executed in pure k8s environments. 
I also have doubts if the topolvm operator repository is the right place to develop it, I think the tool  has enough utility and importance to have  his own repository, and to work as a separate project.

Can you clarify what is going to be the roadmap for this tool?
Why do you think we need to have it inside topolvm ( as a separate tool that dumps results in a directory) ?

**Comment by nbalacha** [comment]
> If we are going to use this tool from inside topolvm, I think we need a better way to retrieve the information directly from the "code", I mean.. we will need to do several changes to use the tool as library. In other side, it is for openshift... now we have a very good opportunity to make it more generic, allowing to be executed in pure k8s environments. I also have doubts if the topolvm operator repository is the right place to develop it, I think the tool has enough utility and importance to have his own repository, and to work as a separate project.
> 
> Can you clarify what is going to be the roadmap for this tool? Why do you think we need to have it inside topolvm ( as a separate tool that dumps results in a directory) ?


Hi Juan,

The must-gather command is specific to Openshift and required for debugging. 
More information about it is available at : https://docs.openshift.com/container-platform/4.4/support/gathering-cluster-data.html

Each openshift operator provides a must-gather utility which is currently implemented in each operator's code base. There have been discussions about moving them all to separate must-gather repo, but until that is done we will have it as part of lvm-operator. We will be building and shipping a separate lvm-must-gather image.

File: must-gather/README.md (hunk 2 above)
Comment (nbalacha) [reply must-gather/README.md:4]: This will be lvm-must-gather

**Comment by jmolmo** [comment]
> The must-gather command is specific to Openshift and required for debugging.

Thanks for the clarification @nbalacha , lot of things are completely new for me.. sorry for the dumb questions

**Review by jmolmo — APPROVED** [review]

File: must-gather/Centos-Base.repo (hunk 1 above)
Comment (nbalacha) [reply must-gather/Centos-Base.repo:2]: Looks like this is not required at all.  I will remove it.

**Comment by nbalacha** [comment]
Not dumb at all. A lot of this is new to many of us.

**Review by leelavg — APPROVED** [review]
/lgtm

**Pull request merged**